ui/src/app/applications/components/application-sync-panel/application-sync-panel.tsx

```diff
@@ -172,9 +172,28 @@ export const ApplicationSyncPanel = ({application, selectedResource, hide}: {app
                                                 .filter(item => !item.hook)
                                                 .map((item, i) => {
                                                     const resKey = nodeKey(item);
+                                                    const contentStart = resKey.substr(0, Math.floor(resKey.length / 2));
+                                                    let contentEnd = resKey.substr(-Math.floor(resKey.length / 2));
+                                                    // We want the ellipsis to be in the middle of our text, so we use RTL layout to put it there.
+                                                    // Unfortunately, strong LTR characters get jumbled around, so make sure that the last character isn't strong.
+                                                    const indexOfFirstLetter = /[a-z]/i.exec(contentEnd).index;
+                                                    contentEnd = contentEnd.slice(indexOfFirstLetter);
                                                     return (
                                                         <div key={resKey} className='application-sync-panel__resource'>
-                                                            <CheckboxField id={resKey} field={`resources[${i}]`} /> <label htmlFor={resKey}>{resKey}</label>{' '}
+                                                            <CheckboxField id={resKey} field={`resources[${i}]`} />
+                                                            <Tooltip content={<div style={{wordBreak: 'break-all'}}>{resKey}</div>}>
+                                                                <div className='container'>
+                                                                    <label htmlFor={resKey} style={{display: resKey.length <= 68 ? '' : 'none'}}>
```

Comment:
Instead of setting the `display` property of these elements conditionally, I think it would be simpler to do something like: 

```js
isLongLabel = resKey.length > 68;

...

<label htmlFor={resKey} content-start={isLongLabel && contentStart} content-end={isLongLabel && contentEnd} />
```

Also, I'm can't find documentation about the `content-start` or `content-end` properties. Could you explain what they do?

Reply:
They're custom things which are used by the css.

I'm not sure I'm a huge fan of the names, although I don't offhand have a suggestion for better names.

Reply:
@rbreeze Good point on the inline style. I'm removing it. Please confirm updated PR. Thanks for the review.